Add integration test for kafka reactive messaging

geoandri opened the pull request:
Closes #12139

geoandri committed:
Add integration test for kafka reactive messaging

machi1990 commented on integration-tests/reactive-messaging-kafka/src/main/java/io/quarkus/it/kafka/PeopleManager.java:
```diff
@@ -0,0 +1,27 @@
+    private final List<Person> list = new CopyOnWriteArrayList<>();
+
+    @Incoming("people-in")
+    public void consume(String json) {
+        Person person = new JsonObject(json).mapTo(Person.class);
+        list.add(person);
+    }
```
```suggestion
    public void consume(Person json) {
        list.add(person);
    }
```

Good PR. Should we use the value deserializer since we have it in config?

geoandri replied:
Yes, you are right.

machi1990 commented on integration-tests/pom.xml:
```diff
@@ -147,6 +147,7 @@
         <module>grpc-health</module>
         <module>google-cloud-functions-http</module>
         <module>google-cloud-functions</module>
+        <module>reactive-messaging-kafka</module>
```
We'll have to add the IT job in the CI too, especially the native part. It can go here https://github.com/quarkusio/quarkus/blob/master/.github/workflows/ci-actions.yml#L476

geoandri replied:
Added to the latest commit.

geoandri commented:
Many thanks, @machi1990 for reviewing this.
 
I have addressed your comments on the latest commit. My main problem though is that the test fails and I think I am missing something crucial. I thought initially that it might be something related to my environment but the test fails in CI as well. I could use some help with this.

machi1990 commented on integration-tests/reactive-messaging-kafka/pom.xml:
```diff
@@ -0,0 +1,297 @@
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-integration-test-reactive-messaging-kafka</artifactId>
```
We can also add name and description following this pattern:

https://github.com/quarkusio/quarkus/blob/efc6dc8e43599d3d574c468fdf70a02e1f226acb/integration-tests/reactive-messaging-amqp/pom.xml#L14-L15

geoandri replied:
Added to the latest commit.

machi1990 commented on integration-tests/reactive-messaging-kafka/src/main/resources/application.properties:
```diff
@@ -0,0 +1,13 @@
+mp.messaging.outgoing.people-out.connector=smallrye-kafka
+mp.messaging.outgoing.people-out.topic=people
+mp.messaging.outgoing.people-out.value.serializer=io.quarkus.kafka.client.serialization.JsonbSerializer
+mp.messaging.incoming.people-in.connector=smallrye-kafka
+mp.messaging.incoming.people-in.value.deserializer=io.quarkus.it.kafka.PersonDeserializer
```
I think the incoming sink is missing the topic.
```suggestion
mp.messaging.incoming.people-in.connector=smallrye-kafka
mp.messaging.incoming.people-in.value.deserializer=io.quarkus.it.kafka.PersonDeserializer
mp.messaging.incoming.people-in.topic=people
```

geoandri replied:
Added the topic for the incoming sink. I had used as a guide the example project (https://quarkus.io/guides/kafka#configuring-the-kafka-connector) which does not actually set the topic for the incoming sink.

machi1990 commented:
> Many thanks, @machi1990 for reviewing this.
> 
> I have addressed your comments on the latest commit. My main problem though is that the test fails and I think I am missing something crucial. I thought initially that it might be something related to my environment but the test fails in CI as well. I could use some help with this.

Thanks for taking care of my suggestions. Do you happen to have a stacktrace that contains any errors? 

I added a suggestion about missing topic name for the incoming sink. Hopeful that should fix the build failure. 

/cc @cescoffier you might have a better idea than mine.

geoandri commented:
Hi @machi1990, 

adding the topic name for the incoming sink did not resolve the issue. The failure of the test is due to the timeout of the `await` method. Seems like nothing is published in Kafka. The stacktrace is 

`[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.it.kafka.KafkaConnectorTest
2020-10-15 07:10:23,445 WARN  [kaf.ser.BrokerMetadataCheckpoint] (main) No meta.properties file under dir /home/geoandri/projects/quarkus/integration-tests/reactive-messaging-kafka/target/data/kafka-data/kafka/broker1/meta.properties
2020-10-15 07:10:23,894 INFO  [io.deb.kaf.KafkaServer] (main) Started Kafka server 1 at localhost:19092 with storage in /home/geoandri/projects/quarkus/integration-tests/reactive-messaging-kafka/target/data/kafka-data/kafka/broker1
2020-10-15 07:10:24,285 INFO  [io.sma.rea.mes.provider] (main) SRMSG00208: Deployment done... start processing
2020-10-15 07:10:24,286 INFO  [io.sma.rea.mes.provider] (main) SRMSG00226: Found incoming connectors: [smallrye-kafka]
2020-10-15 07:10:24,286 INFO  [io.sma.rea.mes.provider] (main) SRMSG00227: Found outgoing connectors: [smallrye-kafka]
2020-10-15 07:10:24,287 INFO  [io.sma.rea.mes.provider] (main) SRMSG00229: Channel manager initializing...
2020-10-15 07:10:24,294 INFO  [io.sma.rea.mes.kafka] (main) SRMSG18200: Merging config with {bootstrap.servers=localhost:19092}
2020-10-15 07:10:24,296 WARN  [io.sma.rea.mes.kafka] (main) SRMSG18216: No `group.id` set in the configuration, generate a random id: 6a3c0262-8041-4050-86d0-03f54ad131ea
2020-10-15 07:10:24,298 INFO  [io.sma.rea.mes.kafka] (main) SRMSG18229: Consumed topics for channel 'people-in': [people]
2020-10-15 07:10:24,301 INFO  [io.sma.rea.mes.kafka] (main) SRMSG18214: Key deserializer omitted, using String as default
2020-10-15 07:10:24,348 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'group.id' was supplied but isn't a known config.
2020-10-15 07:10:24,368 INFO  [io.sma.rea.mes.kafka] (main) SRMSG18200: Merging config with {bootstrap.servers=localhost:19092}
2020-10-15 07:10:24,370 INFO  [io.sma.rea.mes.kafka] (main) SRMSG18214: Key deserializer omitted, using String as default
2020-10-15 07:10:24,394 INFO  [io.sma.rea.mes.provider] (main) SRMSG00209: Initializing mediators
2020-10-15 07:10:24,399 INFO  [io.sma.rea.mes.provider] (main) SRMSG00215: Connecting mediators
2020-10-15 07:10:24,400 INFO  [io.sma.rea.mes.provider] (main) SRMSG00216: Attempt to resolve io.quarkus.it.kafka.PeopleManager#consume
2020-10-15 07:10:24,400 INFO  [io.sma.rea.mes.provider] (main) SRMSG00217: Connecting io.quarkus.it.kafka.PeopleManager#consume to `[people-in]` (org.eclipse.microprofile.reactive.streams.operators.core.PublisherBuilderImpl@369daa11)
2020-10-15 07:10:24,422 INFO  [io.sma.rea.mes.provider] (main) SRMSG00220: Connecting method io.quarkus.it.kafka.PeopleProducer#generatePeople to sink people-out
2020-10-15 07:10:24,474 WARN  [org.apa.kaf.cli.NetworkClient] (vert.x-kafka-consumer-thread-0) [Consumer clientId=consumer-6a3c0262-8041-4050-86d0-03f54ad131ea-1, groupId=6a3c0262-8041-4050-86d0-03f54ad131ea] Error while fetching metadata with correlation id 2 : {people=LEADER_NOT_AVAILABLE}
2020-10-15 07:10:24,474 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | producer-1) [Producer clientId=producer-1] Error while fetching metadata with correlation id 3 : {people=LEADER_NOT_AVAILABLE}
2020-10-15 07:10:24,523 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 2.972s. Listening on: http://0.0.0.0:8081
2020-10-15 07:10:24,524 INFO  [io.quarkus] (main) Profile test activated. 
2020-10-15 07:10:24,524 INFO  [io.quarkus] (main) Installed features: [cdi, mutiny, resteasy, resteasy-jsonb, smallrye-context-propagation, smallrye-health, smallrye-reactive-messaging, smallrye-reactive-messaging-kafka, vertx]
2020-10-15 07:10:24,579 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | producer-1) [Producer clientId=producer-1] Error while fetching metadata with correlation id 4 : {people=LEADER_NOT_AVAILABLE}
2020-10-15 07:10:24,579 WARN  [org.apa.kaf.cli.NetworkClient] (vert.x-kafka-consumer-thread-0) [Consumer clientId=consumer-6a3c0262-8041-4050-86d0-03f54ad131ea-1, groupId=6a3c0262-8041-4050-86d0-03f54ad131ea] Error while fetching metadata with correlation id 25 : {people=LEADER_NOT_AVAILABLE}
2020-10-15 07:10:24,952 INFO  [io.qua.it.kaf.KafkaTestResource] (main) Test run
2020-10-15 07:10:25,663 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,044 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,188 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,317 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,446 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,600 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,738 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:26,877 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,014 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,148 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,281 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,414 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,547 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,679 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,813 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:27,970 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,099 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,230 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,351 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,474 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,603 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,730 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,854 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:28,970 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,079 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,200 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,316 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,434 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,560 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,679 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,806 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:29,927 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,050 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,176 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,300 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,422 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,541 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,668 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,785 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:30,892 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,002 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,123 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,253 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,380 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,501 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,620 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,739 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,863 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:31,985 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,100 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,211 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,332 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,449 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,564 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,687 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,803 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:32,919 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,045 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,169 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,291 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,408 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,523 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,632 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,768 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:33,889 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,009 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,129 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,248 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,369 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,489 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,602 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,716 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,838 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
2020-10-15 07:10:34,958 INFO  [io.qua.it.kaf.PeopleManager] (executor-thread-1) Returning people []
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 15.333 s <<< FAILURE! - in io.quarkus.it.kafka.KafkaConnectorTest
[ERROR] io.quarkus.it.kafka.KafkaConnectorTest.test  Time elapsed: 10.456 s  <<< FAILURE!
org.opentest4j.AssertionFailedError
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:47)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:140)
	at io.quarkus.it.kafka.KafkaConnectorTest.test(KafkaConnectorTest.java:32)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:764)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:671)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: org.awaitility.core.ConditionTimeoutException: Condition with io.quarkus.it.kafka.KafkaConnectorTest was not fulfilled within 10 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:864)
	at io.quarkus.it.kafka.KafkaConnectorTest.test(KafkaConnectorTest.java:29)
	... 70 more
Caused by: java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:204)
	at org.awaitility.core.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:101)
	at org.awaitility.core.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:81)
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:101)
	... 75 more

2020-10-15 07:10:35,078 INFO  [io.sma.rea.mes.provider] (main) SRMSG00207: Cancel subscriptions
2020-10-15 07:10:35,086 INFO  [io.quarkus] (main) Quarkus stopped in 0.078s
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   KafkaConnectorTest.test:32
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
`

cescoffier commented on integration-tests/reactive-messaging-kafka/src/main/java/io/quarkus/it/kafka/PeopleManager.java:
```diff
@@ -0,0 +1,24 @@
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.jboss.logging.Logger;
+
+public class PeopleManager {
```
Add @ApplicationScoped

cescoffier commented on .github/workflows/ci-actions.yml:
```diff
@@ -481,6 +481,7 @@ jobs:
               kafka
               kafka-streams
               reactive-messaging-amqp
+              reactive-messaging-kafka
```
You should add 10 minutes to the timeout.

cescoffier commented on integration-tests/reactive-messaging-kafka/src/test/java/io/quarkus/it/kafka/KafkaConnectorTest.java:
```diff
@@ -0,0 +1,37 @@
+    public void test() {
+        log.info("Test run");
+        try {
+            await()
```
I would use `await().untilAsserted`

cescoffier replied:
```java
@Test
public void test() {
        await().untilAsserted(() -> 
           Assertions.assertEquals(get("/kafka/people").as(TYPE_REF).size(), 6));
}
```

cescoffier commented on integration-tests/reactive-messaging-kafka/src/main/resources/application.properties:
```diff
@@ -0,0 +1,14 @@
+mp.messaging.outgoing.people-out.topic=people
+mp.messaging.outgoing.people-out.value.serializer=io.quarkus.kafka.client.serialization.JsonbSerializer
+mp.messaging.incoming.people-in.connector=smallrye-kafka
+mp.messaging.incoming.people-in.topic=people
```
The tests fail because of a missing property:
```
mp.messaging.incoming.people-in.auto.offset.reset=earliest
```

Without, it uses Kafka's. default which is _latest_. Basically, it only receives records sends after the subscription. _earliest_ instructs the consumers to _reset_ its offset to the last processed offset from consumers from its consumer group. Here, it will reset the offset to 0, as there are no other consumers (so obviously none in its group) and no records have been erased due to retention expiration (24h by default).

geoandri replied:
Indeed, many thanks @cescoffier for the help and the detailed explanation as well.

cescoffier commented on integration-tests/reactive-messaging-kafka/src/test/java/io/quarkus/it/kafka/KafkaConnectorTest.java:
```diff
@@ -0,0 +1,37 @@
+
+@QuarkusTest
+@QuarkusTestResource(KafkaTestResource.class)
+public class KafkaConnectorTest {
```
Could you add the associated native tests?

```java
package io.quarkus.it.kafka;

import io.quarkus.test.junit.NativeImageTest;

@NativeImageTest
public class KafkaConnectorIT extends KafkaConnectorTest {

}
```

geoandri replied:
Done

cescoffier reviewed (changes requested):
I've found why the tests are failing. See my comments.

machi1990 commented:
@geoandri thanks for taking into account our suggestions. Do you mind squashing the commits into one commit?

geoandri commented:
Squashed the commits @machi1990 . Thank you.

machi1990 reviewed (approved):
LGTM. Thanks @geoandri

cescoffier commented:
Thanks! It was on my todo list for a very long time.

geoandri commented:
You're welcome @cescoffier .
